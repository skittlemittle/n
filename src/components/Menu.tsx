import styled from "styled-components";

const MenuBox = styled.div.attrs((props: { top: number; left: number }) => ({
  top: props.top,
  left: props.left,
}))`
  position: absolute;
  top: ${(props) => props.top}px;
  left: ${(props) => props.left}px;
  min-width: 80px;
  padding: 0.25em;
  background: ${(props) => props.theme.colors.bg0};
  border: 1px solid ${(props) => props.theme.colors.bg4};
  border-radius: 2px;
  box-shadow: 1px 1px 6px 1px rgba(124, 111, 100, 0.21);
`;

const MenuItem = styled.div`
  color: ${(props) => props.theme.colors.fg1};
  font-family: "Inter";
  font-style: normal;
  font-size: 14px;
  padding: 0.35em;

  &:hover {
    background: ${(props) => props.theme.colors.bg0_h};
  }
`;

interface Item {
  text: string;
  action: () => void;
}

interface Props {
  items: Item[];
  top: number;
  left: number;
}

const Menu = (props: Props) => {
  const handleClick = (e: React.MouseEvent, action: () => void) => {
    e.preventDefault();
    e.stopPropagation();
    e.nativeEvent.stopImmediatePropagation();
    action();
  };

  return (
    <MenuBox top={props.top} left={props.left}>
      {props.items.map((it, i) => (
        <MenuItem
          key={i}
          onClick={(e: React.MouseEvent) => handleClick(e, it.action)}
        >
          {it.text}
        </MenuItem>
      ))}
    </MenuBox>
  );
};

export default Menu;
