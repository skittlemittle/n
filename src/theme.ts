import { COLORS_L } from "./themeConstants";

export const AppTheme = {
  colors: COLORS_L,
};
