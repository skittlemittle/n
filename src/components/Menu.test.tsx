import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import { AppTheme } from "../theme";
import Menu from "./Menu";

it("menu style", () => {
  const m = renderer
    .create(
      <ThemeProvider theme={AppTheme}>
        {" "}
        <Menu
          items={[
            {
              text: "owo",
              action: () => {},
            },
            {
              text: "big chungus zone",
              action: () => {},
            },
            {
              text: "big chungus zone",
              action: () => {},
            },
            {
              text: "big chungus zone",
              action: () => {},
            },
          ]}
          top={200}
          left={200}
        />
      </ThemeProvider>
    )
    .toJSON();
  expect(m).toMatchSnapshot();
});
