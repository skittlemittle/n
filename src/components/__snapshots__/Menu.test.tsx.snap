// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`menu style 1`] = `
Array [
  " ",
  <div
    className="sc-bdvvtL dEZTpU"
  >
    <div
      className="sc-gsDKAQ jhKwWD"
      onClick={[Function]}
    >
      owo
    </div>
    <div
      className="sc-gsDKAQ jhKwWD"
      onClick={[Function]}
    >
      big chungus zone
    </div>
    <div
      className="sc-gsDKAQ jhKwWD"
      onClick={[Function]}
    >
      big chungus zone
    </div>
    <div
      className="sc-gsDKAQ jhKwWD"
      onClick={[Function]}
    >
      big chungus zone
    </div>
  </div>,
]
`;
